ql/src/java/org/apache/hadoop/hive/ql/exec/MapJoinOperator.java

```diff
@@ -556,6 +556,9 @@ public void process(Object row, int tag) throws HiveException {
               }
             } else {
               storage[pos] = emptyList;
+              if (pos != 0 && condn[pos - 1].getType() == JoinDesc.ANTI_JOIN) {
```

Comment:
Can we add a comment here?  Thanks!

Reply:
Thank you for your review. I added a comment which explains the reason for proceeding join.

Reply:
Just for reference, this check matches the check that is done in CommonJoinOperator.checkAndGenObject() here: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/CommonJoinOperator.java#L973 .  Looks good to me.